sympy/solvers/ode.py

```diff
@@ -6409,28 +6408,26 @@ def ode_lie_group(eq, func, order, match):
     except NotImplementedError:
         eqsol = []
 
-    desols = []
+    desols = set()
     for s in eqsol:
         sol = _ode_lie_group(s, func, order, match=match)
         if sol:
-            desols.extend(sol)
+            desols.update(sol)
 
-    if desols == []:
+    if not desols:
         raise NotImplementedError("The given ODE " + str(eq) + " cannot be solved by"
             + " the lie group method")
-    return desols
+    return list(ordered(desols))
```

Comment:
These duplicates are supposed to be handled later on by `_remove_redundant_solutions`. It seems that that function doesn't handle the case where the solutions are completely identical. It works if they have different symbols:
```julia
In [5]: eq = (f(x).diff(x)**2 - f(x)**3)                                                                                                       

In [6]: sol = dsolve(eq)                                                                                                                       

In [7]: from sympy.solvers.ode import _remove_redundant_solutions                                                                              

In [8]: _remove_redundant_solutions(eq, sol, 1, f(x))                                                                                          
Out[8]: 
⎡               4                         4        ⎤
⎢f(x) = ─────────────────, f(x) = ─────────────────⎥
⎢         2             2           2             2⎥
⎣       C₁  + 2⋅C₁⋅x + x          C₁  + 2⋅C₁⋅x + x ⎦

In [9]: sol2 = [sol[0], sol[1].subs(Symbol('C1'), Symbol('C2'))]                                                                               

In [10]: sol2                                                                                                                                  
Out[10]: 
⎡               4                         4        ⎤
⎢f(x) = ─────────────────, f(x) = ─────────────────⎥
⎢         2             2           2             2⎥
⎣       C₁  + 2⋅C₁⋅x + x          C₂  + 2⋅C₂⋅x + x ⎦

In [12]: _remove_redundant_solutions(eq, sol2, 1, f(x))                                                                                        
Out[12]: 
⎡               4        ⎤
⎢f(x) = ─────────────────⎥
⎢         2             2⎥
⎣       C₁  + 2⋅C₁⋅x + x ⎦
```
We can fix this with
```
diff --git a/sympy/solvers/ode.py b/sympy/solvers/ode.py
index 6901e13974..41e7d15aac 100644
--- a/sympy/solvers/ode.py
+++ b/sympy/solvers/ode.py
@@ -4730,7 +4730,7 @@ def _is_special_case_of(soln1, soln2, eq, order, var):
     constants1 = soln1.free_symbols.difference(eq.free_symbols)
     constants2 = soln2.free_symbols.difference(eq.free_symbols)
 
-    constants1_new = get_numbered_constants(soln1 - soln2, len(constants1))
+    constants1_new = get_numbered_constants(Tuple(soln1, soln2), len(constants1))
     if len(constants1) == 1:
         constants1_new = {constants1_new}
     for c_old, c_new in zip(constants1, constants1_new):
```
That ensures that both sides get substituted with new symbols. The problem is that `soln1 - soln2` just gives zero so we don't get new constants to substitute.